### PR TITLE
Fix Auth Email Confirmation Flow

### PR DESCRIPTION
**Pull request by meganniu**
Flow:
+ User creates account
+ Confirmation email sent to user
+ User confirms their email using the confirmation link
+ User can successfully login

https://user-images.githubusercontent.com/20423590/113874428-c960b400-9783-11eb-9a22-4346728d7888.mov

Flow:
+ User creates account
+ Confirmation email sent to user
+ User does NOT confirm their email address
+ User tries to login
+ Another confirmation email is sent to user on the login attempt

https://user-images.githubusercontent.com/20423590/113874444-cc5ba480-9783-11eb-89af-a8eeba9cd42d.mov

### CODE REVIEW

File: client/src/pages/SignInModal.tsx
Comment (meganniu) [inline client/src/pages/SignInModal.tsx:50]: Sorry there's a big diff here! I wrapped everything in `React.Fragment`, so everything was reindented.

My only changes here were:
+ wrap everything in `React.Fragment`
+ show ConfirmationModal if a confirmation email was resent

**Review by Anand-Issac — APPROVED** [review]
lgtm 👍

File: client/src/services/auth.ts
Comment (chamod-gamage) [inline client/src/services/auth.ts]: We'll want to use AuthErrorMessage["unconfirmed-email"] instead :)

**Review by chamod-gamage — CHANGES_REQUESTED** [review]
One small change but everything else looks good to me! Would wait until #104 is merged due to changes to SignIn/Up modal styling.

File: client/src/services/auth.ts
Comment (meganniu) [reply client/src/services/auth.ts]: gotchu

**Review by chamod-gamage — APPROVED** [review]
Awesome work - LGTM!

**Pull request merged**
